src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/annotations/PostConstructReturnTypeQuickFix.java

```diff
@@ -72,7 +73,7 @@ public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
 
         assert parentType != null;
         ChangeCorrectionProposal proposal = new ModifyReturnTypeProposal(TITLE_MESSAGE, context.getSource().getCompilationUnit(),
-                context.getASTRoot(), parentType, 0, PsiPrimitiveType.VOID);
+                context.getASTRoot(), parentType, 0, PsiTypes.voidType());
```

Comment:
What is the difference between the old, deprecated methods PsiPrimitiveType and PsiType? I see that you changed the code to replace both deprecated methods with the PsiTypes method.

Answer: @vaisakhkannan 
Both are Jetbrain's classes.
PsiPrimitiveType class - Representation of Java type (primitive type, array or class type).

**Deprecated constants**
All static fields in PsiPrimitiveType class representing instances of PsiPrimitiveType are deprecated. It was done to avoid deadlocks during initialization of the class. According to section 5.5  of JVM specification, when a class is initialized, JVM firsly synchronizes on an initialization lock specific for that class, then initializes its super class and then computes initializers for its static fields. So because of these fields initialization of PsiType performs initialization of PsiPrimitiveType, and initialization of PsiPrimitiveType performs initialization of PsiType because its the super class of its super class. Therefore, if one thread starts initialization of PsiType, and another thread starts initialization of PsiPrimitiveType at the same time, it'll result in a deadlock. In order to avoid this, methods from PsiTypes must be used to get instances of the primitive types.
As suggested by Jetbrains we are using PsiTypes methods to get the PsiPrimititveType constants, instead of accessing them directly.

This how PsiTypes method works.

public static @NotNull PsiPrimitiveType voidType() { 
    return VOID; 
  }

Reply:
Thanks @aparnamichael for the explanation.